design/pipeline-artifacts.md

```diff
@@ -37,23 +39,80 @@ Artifacts are distinct from other `resources` types, including `containers`, `re
 
 `download` is a shortcut for the [Download Pipeline Artifacts](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-pipeline-artifact) task.
 
-It will download artifacts uploaded from a previous job or stage or from another pipeline.
+It will download artifacts uploaded from a previous job, stage, or from another pipeline.
 
 ### Artifact download location
 
-Artifacts are downloaded either to `$PIPELINE_RESOURCESDIRECTORY` or to the directory specified in `root`. Each artifact is given its own directory e.g. `$PIPELINE_RESOURCESDIRECTORY\default` for the `default` artifact of this pipeline. Artifacts coming from other Pipelines are each given one directory per pipeline e.g. `$PIPELINE_RESOURCESDIRECTORY\some-other-pipeline\default` for the `default` artifact of the `some-other-pipeline` pipeline.
+#### YAML / unified pipelines
+Artifacts are downloaded to the pipeline's workspace by default. We introduce a new variable for unified pipelines: `$(Pipeline.Workspace)`. `Pipeline.Workspace` is one level up from `$(System.DefaultWorkingDirectory)`; on hosted, it corresponds to `c:\agent\_work\1\`.
+
+Each artifact is given its own directory e.g. `$(Pipeline.Workspace)\myartifact` for an artifact named `myartifact`. Artifacts coming from other pipelines are each given one directory per pipeline e.g. `$(Pipeline.Workspace)\some-other-pipeline\someartifact` for the `someartifact` artifact of the `some-other-pipeline` pipeline. The pipeline portion of the path comes from the ID of the pipeline's `resource` in YAML.
+
+Example:
+```yaml
+resources:
+  pipelines:
+  - pipeline: mypipe
+
+jobs:
+- job: makeartifact
+  steps:
+  - script: ./build.sh
+  - upload: outputs/**/*
+
+- job: use1artifact
+  dependsOn: makeartifact
+  steps:
+  # by default, a download step is injected -- see later section
+  - script: ls $(Pipeline.Workspace)
+    # this listing shows a folder called `makeartifact`, since the prior job didn't specify a name
+
+- job: use2artifacts
+  dependsOn: makeartifact
+  steps:
+  - download: mypipe   # downloads all artifacts from `mypipe`
+  - download: current  # must include this, since by including a download step, we don't get automatic behavior anymore
```

Comment:
We need to make sure we reserve the alias current in the system.

Reply:
Yeah, we should make it work like `self`. I don't know where or if that's reserved.

Reply:
Any reason why we wouldn't re-use ```self``` as the way to identify?

Reply:
`self` refers to the repo. `current` refers to the pipeline run.

Reply:
I don't remember all the discussion around this - but we had some concerns about making `self` ambiguous.

Reply:
OK - just making sure there was a reason to be different, sounds like there is.